know/rdf/parser/wikidata_parser.py

```diff
@@ -1,26 +1,44 @@
 # pylint: disable=no-self-use,unused-import,
 import re
+from qwikidata.sparql import return_sparql_query_results
 from rdf.parser.abstract_parser import AbstractParser, EntityType
+from rdf.parser.sparql_reader import read_sparql
 class WikidataParser(AbstractParser):
     """parses wikidata sources"""
 
     def __init__(self, uri):
         super().__init__(uri)
-        self.entity_id = re.search('Q[0-9]+', self.uri)
+        self.entity_id = re.search('Q[0-9]+', self.uri).group(0)
         if self.entity_id is None:
             raise Exception("Invalid Wikidata URI")
 
     def parse(self) -> dict:
         """call this method to start parsing"""
-        return dict()
+        entity_type =  self.get_entity_type()
+        if entity_type == EntityType.BOOK:
+            return self.parse_book()
+        if entity_type == EntityType.PERSON:
+            return self.parse_person()
+        raise Exception("Unsupported entity type!")
     def get_entity_type(self) -> str:
         "gets the entity type"
-        return ""
+        query = read_sparql("get_instance.sparql", self.entity_id)
+        response =  return_sparql_query_results(query)
+        for i in response['results']['bindings']:
```

Comment:
```suggestion
        for i in response.get('results').get('bindings'):
```
We don't need to entirely add a bunch of sanity checks, but `dict`s throw an error if there isn't a key that matches when you use the `['key']` syntax, but `.get()` just returns `None`. As such, use `get` in its place. (although it'll probably throw an error if `'results'` is `None`, but w/e it's still better than it throwing an error for both of them not existing)